astropy/units/format/base.py

```diff
@@ -208,3 +207,111 @@ def parse(cls, s: str) -> UnitBase:
         Convert a string to a unit object.
         """
         raise NotImplementedError(f"Can not parse with {cls.__name__} format")
+
+
+class ParsingFormatBase(Base):
+    """Extends Base with basic building blocks for parsing units."""
+
+    _deprecated_units: ClassVar[frozenset[str]] = frozenset()
+
+    @classmethod
+    def parse(cls, s: str, debug: bool = False) -> UnitBase:
+        try:
+            return cls._parser.parse(s, lexer=cls._lexer, debug=debug)
+        except ValueError as e:
+            if str(e):
+                raise
+            else:
+                raise ValueError(f"Syntax error parsing unit '{s}'")
```

Comment:
Raise from or from None?

```suggestion
                raise ValueError(f"Syntax error parsing unit '{s}'") from e
```

Reply:
Resolved.